helper_functions.js

```diff
@@ -72,4 +25,180 @@ module.exports = {
     );
     return difference <= 10;
   },
+
+  /**
+   * Gets the number of the week a certain date is in. The calculations are done from an original week, a week
+   * that our system knows both the start date and week number of.
+   *
+   * @param {*} date the date to get the week number of.
+   * @param {*} originalWeekStartDate the start date of the original week.
+   * @param {*} originalWeekNumber the number of the original week.
+   * @returns the number of the week this date is in.
+   */
+  getWeekNumberForDate: function (
+    date,
+    originalWeekStartDate,
+    originalWeekNumber
+  ) {
+    // Split the originalWeekStartDate parameter into day, month and year
+    const splittedoriginalWeekStartDate = originalWeekStartDate.split("/");
+    const originalDay = splittedoriginalWeekStartDate[1];
+    const originalMonth = splittedoriginalWeekStartDate[0];
+    const originalYear = splittedoriginalWeekStartDate[2];
+    // Split the date parameter into day, month and year
+    const splittedDate = date.split("/");
+    const requestedDay = splittedDate[1];
+    const requestedMonth = splittedDate[0];
+    const requestedYear = splittedDate[2];
+
+    // The days difference between the given date and today
+    const difference = getDifference(
+      originalDay,
+      originalMonth,
+      originalYear,
+      requestedDay,
+      requestedMonth,
+      requestedYear
+    );
+
+    const numberOfWeeksElapsed = difference / 7;
+    return originalWeekNumber + numberOfWeeksElapsed;
+  },
+
+  /**
+   * @param {*} date the date under MM/DD/YYY format.
+   * @returns the week day it corresponds to.
+   */
+  getWeekDayFromDate: function (dateString) {
+    const [month, day, year] = dateString.split("/");
+    const date = new Date(+year, month - 1, +day);
+    const weekdays = [
+      "Sunday",
+      "Monday",
+      "Tuesday",
+      "Wednesday",
+      "Thursday",
+      "Friday",
+      "Saturday",
+    ];
+
+    return weekdays[date.getDay()];
+  },
+
+  /**
+   * Checks if date is in the MM/DD/YYYY format.
+   *
+   * @param {*} date the date to test.
+   * @returns if it's in the specified format.
+   */
+  isValidDateFormat: function (date) {
+    if (!date.includes("/")) {
+      return false;
+    }
+    const splittedDate = date.split("/");
+    if (!splittedDate.length === 3) {
+      return false;
+    } else {
+      const numericalDay = parseInt(splittedDate[1]);
+      const numericalMonth = parseInt(splittedDate[0]);
+      if (
```

Comment:
you could make it
```
if (!splittedDate.length === 3) {
      return false;
    } else {
      const numericalDay = parseInt(splittedDate[1]);
      const numericalMonth = parseInt(splittedDate[0]);
      return
        splittedDate[0].length == 2 &&
        splittedDate[1].length == 2 &&
        splittedDate[2].length == 4 &&
        numericalDay < 32 &&
        numericalMonth < 13;
    }
```

Reply:
Yep both are great ideas, thanks! I updated it